docs/developers/how-to-build-plugins.md

```diff
@@ -0,0 +1,262 @@
+# How to build model plugins
+
+The IF is designed to be as composable as possible. This means you can develop your own models and use them in a model pipeline. 
+To help developers write Typescript models to integrate easily into IF, we provide the `ModelPluginInterface` interface. Here's an overview of the stages you need to follow to integrate your model:
+
+- create a Typescript file that implements the `ModelPluginInterface`
+- install the model
+- initialize and invoke the model in your manifest file
+
+
+## The model interface
+
+The `ModelPluginInterface` is structured as follows:
```

Comment:
Considering our recent change of language re plugins instead of model plugins, to avoid confusion I would change the interface as well.

Reply:
yes, we will rename when we update the signature